src/Core/src/Hosting/MauiServiceProvider.cs

```diff
@@ -126,5 +128,13 @@ public MauiServiceProvider(IMauiServiceCollection collection, bool constructorIn
 
 			return match.Constructor.Invoke(paramValues);
 		}
+
+		object? GetServiceCore(Type type)
+		{
+			if (ServiceProviderType.IsAssignableFrom(type))
+				return this;
```

Comment:
Hax for the `IServiceProvider` in the ctor. Works for now, but might need a revist later.

Reply:
How will this work if you are using a factory to provide Other container, like for example the default Microsoft.Extensions.DependencyInjection ServiceCollection and ServiceProvider implementation

Reply:
I just did a test with the Microsoft.Extensions.DependencyInjection package and it also got injected. Was there a specific case that failed?